### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by 1nv8rzim**

### CODE REVIEW

File: commands/scheduled/birthday.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		}
+
+		// Remove yesterday's birthdays
+		for _, entRemoveBirthday := range entRemoveBirthdays {
+			err = removeBirthday(s, entRemoveBirthday.Edges.User.ID, internalSpan.Context())
+			if err != nil {
+				logging.Error(s, "failed to remove birthday for "+helpers.AtUser(entRemoveBirthday.Edges.User.ID), nil, span, logrus.Fields{"error": err})
+				return
+			}
+		}
```
Comment (jabbate19) [inline commands/scheduled/birthday.go:88]: Edge/Exploit Case: I move my birthday on my birthday. I wouldn't come up here

**Review by jabbate19 — CHANGES_REQUESTED** [review]
Suggestion to prevent issue: Prevent change while its your birthday or limit number of changes allowed

File: commands/slash/birthday.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+		entBirthday, err := data.Birthday.Get(i.Member.User.ID, span.Context())
+		if err != nil {
+			logging.Error(s, "encounted error when getting birthday from user", i.Member.User, span, logrus.Fields{"err": err.Error()})
+
+			return
+		}
+
+		// check if birthday is today
+		now := time.Now().In(tx)
+		if int(entBirthday.Month) == int(now.Month()) && int(entBirthday.Day) == int(now.Day()) {
```
Comment (1nv8rzim) [inline commands/slash/birthday.go:153]: Birthday Delete on Birthday Edge Case

File: commands/slash/birthday.go [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+	tz, err := time.LoadLocation("America/New_York")
+	if err != nil {
+		logging.Error(s, "encounted error when loading timezone", i.Member.User, span, logrus.Fields{"err": err.Error()})
+
+		return
+	}
+
+	// check if birthday is today
+	now := time.Now().In(tz)
+	if month == int(now.Month()) && day == int(now.Day()) {
```
Comment (1nv8rzim) [inline commands/slash/birthday.go:247]: Birthday Edit or Add on Birthday Edge Case

File: commands/scheduled/birthday.go (hunk 1 above)
Comment (1nv8rzim) [reply commands/scheduled/birthday.go:88]: This is the scheduled task that adds and removes birthdays at midnight. 

This edge case is covered in the snippets commented below

File: commands/slash/birthday.go [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+				Content: "You cannot set your birthday to be today",
+				Flags:   discordgo.MessageFlagsEphemeral,
+			},
+		})
+		if err != nil {
+			logging.Error(s, "encounted error when responding to user", i.Member.User, span, logrus.Fields{"err": err.Error()})
+		}
+
+		return
+	}
```
Comment (1nv8rzim) [inline commands/slash/birthday.go:262]: Birthday Edit or Add on Birthday Edge Case

**Review by jabbate19 — APPROVED** [review]
:shipit:

**Pull request merged**